client/src/components/pages/application/Signup/index.js

```diff
@@ -1,5 +1,195 @@
 import React from 'react';
+import axios from 'axios';
+import Alert from '../../../common/Alert';
+import Card from '../../../common/card';
+import Typography from '../../../common/Typography';
+import TextField from '../../../common/TextField';
+import Button from '../../../common/Button';
+import signupValidate from '../../../../utils/application/SignupValidation';
+import SelectBox from '../../../application/SelectBox';
+import signupImage from '../../../../assets/images/signup.svg';
+import './index.css';
 
-const SignUp = () => <h1>SignUp</h1>;
+const initialState = {
+  fullName: '',
+  email: '',
+  password: '',
+  passwordConfirmation: '',
+  location: '',
+  message: [],
+};
+
+class SignUp extends React.Component {
+  state = initialState;
+
+  handleSignIn = () => {
+    const { history } = this.props;
+    history.push('/login');
+  };
+
+  throwMessage = (message) => this.setState({ message });
+
+  resetForm = () => {
+    this.setState(initialState);
+  };
+
+  submit = () => {
+    const { message, ...rest } = this.state;
+    signupValidate(rest)
+      .then((result) =>
+        result
+          ? axios
+              .post('/api/v1/signup', rest)
+              .then(() => {
+                this.resetForm();
+                this.throwMessage('SignUp Successfully');
+              })
+              .catch(() => this.throwMessage('Your Email is Exist'))
+          : this.throwMessage('Please Enter Correct Data')
+      )
+      .catch(({ errors }) => this.throwMessage(errors));
+  };
+
+  handleChange = (event) => {
+    const { value, name } = event.target;
+    this.setState(
+      {
+        [name]: value,
+      },
+      () => {
+        const { message, ...rest } = this.state;
+        signupValidate(rest)
+          .then(this.throwMessage(''))
+          .catch(({ errors }) => this.throwMessage(errors));
+      }
```

Comment:
here why you extract message & didn't use it below ?

Reply:
I trying to get the object rest destructuring, without the message to send it to validation and the axios functions,